api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,136 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+// Create takes a request from the Controller layer, formats the request,
+// and calls the AWS ECS API with it. In order to do this, we must:
+//
+// - Generate a unique deployID and fqDeployID
+// - Render a AWS TaskDefinition from the Request
+// - Create an ecs.TaskDefinition object for the Task Definition data
+// - Call the AWS' ECS RegisterTaskDefinition API with the previous steps' object
+// - If the AWS API call is successful, Marshal the request into a []byte
+//   - Instantiate Layer0 models.Deploy object with the marshaled data
+// - Insert entity tags into Layer0 tags database
+// - Return entity object back to Controller layer
+//
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	deployID := generateEntityID(req.DeployName)
+	fqDeployID := addLayer0Prefix(d.Config.Instance(), deployID)
+	familyName := fqDeployID
+
+	taskDefinition, err := d.renderTaskDefinition(req.DeployFile, fqDeployID)
+	if err != nil {
+		return nil, err
+	}
+
+	input := &ecs.TaskDefinition{}
```

Comment:
generating the `input` for the aws call should happen inside the helper function. see [some examples](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/environment_create.go#L125)

Reply:
OK, but this contradicts the '~two helper function" standard we were going for (as I recall...)